twisted/names/test/test_tap.py

```diff
@@ -96,4 +101,21 @@ def test_recursiveConfiguration(self):
                 if isinstance(recurser, Resolver):
                     recurser._parseCall.cancel()
 
+        # On Windows, we need to use a threaded resolver, which leaves trash
+        # lying about that we can't easily clean up. This isn't a good bit of
+        # code. Sorry, abstraction gods.
+        if platform.getType() != 'posix':
+
+            # We want the delayed calls on the reactor, which should be all of
+            # ours from the threaded resolver cleanup
+            from twisted.internet import reactor
+            for x in reactor._newTimedCalls:
+                if _PY3:
+                    self.assertEqual(x.func.__func__,
+                                     ThreadedResolver._cleanup)
+                else:
+                    self.assertEqual(x.func.__func__,
+                                     ThreadedResolver._cleanup.__func__)
+                x.cancel()
```

Comment:
it is not clear why the _newTimedCalls are cancelled ... I guess that they are cancelled because we can not clean them up or we can not wait for them to be done before checking the reactor state ... but it is not  clear